core-api/src/main/java/com/optimizely/ab/config/audience/match/Match.java

```diff
@@ -20,5 +20,5 @@
 
 public interface Match {
     @Nullable
-    Boolean eval(Object attributeValue);
+    Boolean eval(Object conditionValue, Object attributeValue) throws UnexpectedValueTypeException, UnknownValueTypeException;
```

Comment:
Do we need separate exception types for UnexpectedValueTypeException and UnknownValueTypeException?
They're similar and not much gain for additional complexity to differentiate them.

Reply:
This is for backwards compatibility, albeit I can update the names perhaps. The difference is in the logging behavior. One error is communicated as an SDK compatibility issue warning to upgrade on the condition value, while the other is saying that the attribute value type is unknown.

Reply:
The distinction is since the conditionValue comes from the datafile, so if a value is of an "unexpected" type then its likely the result of an out of date SDK. Since the attributeValue is provided at runtime via the user attributes, then we log that the type is "unknown".

Reply:
Got it. We can consider adding some clarifications.

Reply:
I added doc strings for all the exception classes in this package.